server/webapp/WEB-INF/rails.new/config/routes.rb

```diff
@@ -145,7 +145,7 @@
   get "admin/package_definitions/check_connection" => "admin/package_definitions#check_connection", as: :package_definition_check_connection
 
   get "admin/package_repositories/new" => "admin/package_repositories#new", as: :package_repositories_new
-  get "admin/package_repositories/check_connection" => "admin/package_repositories#check_connection", as: :package_repositories_check_connection
+  post "admin/package_repositories/check_connection" => "admin/package_repositories#check_connection", as: :package_repositories_check_connection
```

Comment:
You need to add the header constraint here also see
https://github.com/gocd/gocd/blob/1190b342edf23b8ce8e5dabce41a9ecc29ec9638/server/webapp/WEB-INF/rails.new/config/routes.rb#L294